doc/releases/changelog-dev.md

```diff
@@ -471,6 +471,10 @@
 
 * Updated documentation check to remove duplicate docstring references. [(#7453)](https://github.com/PennyLaneAI/pennylane/pull/7453)
 
+* Improved performance for `qml.clifford_t_decomposition` transform by introducing caching support and changed the
```

Comment:
What's the expected boost up? Could you please include the benchmark in the PR description for future reference?